### PR TITLE
Add GraphQL

### PR DESCRIPTION
**Pull request by karsonkalt**
# In this PR
- Exploring adding 4 new technologies as a replacement to REST
  - GraphQL to replace REST
  - Apollo to handle front end requests
  - Express handling the backend server setup
  - `node-json-db` to keep our db as a JSON file
  
## Proposed Structure
```
└── src
    ├── ...
    └── server
        ├── actions
        │   ├── ...
        │   ├── sessions.js
        │   ├── tweets.js
        │   └── users.js
        ├── db
        │   ├── db.json
        │   └── index.js
        ├── index.js
        └── schema
            ├── resolvers.js
            └── typeDefs.js
```

### `/`
Screen Shot 2022-06-06 at 8 20 38 AM

### `/user/:userId`
Screen Shot 2022-06-06 at 8 20 56 AM

### CODE REVIEW

File: src/server/schema/typeDefs.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const { gql } = require('apollo-server-express');
```
Comment (JustKong) [inline src/server/schema/typeDefs.js:1]: This honestly looks pretty solid to me!

Honestly, a lot of the Facebook patterns I'm used to aren't as applicable with this Apollo Server, as we were using Hack for our backend server. The fact that you can so tightly couple the schema and the resolvers here is something that just wasn't possible, or at least something that we chose not to do.

I started looking into this and found this blog post: https://www.apollographql.com/blog/backend/schema-design/modularizing-your-graphql-schema-code/

Based off the blog post, they do advocate to have both the schema and the resolver for a particular query in the same file. That may be a little too much given the complexity of the project so far, but worth thinking about!

Comment (karsonkalt) [reply src/server/schema/typeDefs.js:1]: Fantastic read, thanks for sharing, and I will be implementing this approach!

**Review by badgerbadgers — APPROVED** [review]
LGTM!

**Review by viniciuspietscher — APPROVED** [review]
:+1: :grin:

File: src/server/schema/typeDefs.js (hunk 1 above)
Comment (asaiapalacios) [reply src/server/schema/typeDefs.js:1]: Seeing this example (this discussion between the two) helps me better understand ways I can provide feedback and resources when doing a collaborating review. Thanks.

**Review by asaiapalacios — APPROVED** [review]
Thumbs up.

**Pull request merged**